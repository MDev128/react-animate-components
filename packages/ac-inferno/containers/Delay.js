// @flow

import Inferno from 'inferno';
import Component from 'inferno-component';

type DefaultProps = {
  timeout: number,
  children?: Element<any> // Workaround ðŸ˜•
}

type Props = {
  timeout: number,
  children?: Element<any>
}

type State = {
  show: boolean
}

export default class Delay extends Component<DefaultProps, Props, State> {
  timer = null;

  static displayName = 'Delay';

  static defaultProps = {
    timeout: 4, // Minimum timeout is 4ms so no difference when compared with timeout zero
  };

  state = {
    show: false,
  };

  componentWillMount = () => {
    this.timer = null;
  };

  componentDidMount = () => {
    const { timeout } = this.props;
    this.timer = setTimeout(this.setShowValue, timeout);
  };

  componentWillUnmount = () => {
    clearTimeout(this.timer);
  };

  setShowValue = () => {
    this.setState({
      show: true,
    });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;
    const performAnimation = show ? children : <div />;

    return performAnimation;
  }
}
