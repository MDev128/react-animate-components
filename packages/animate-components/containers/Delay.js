import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Delay extends PureComponent {
  // avoids warning for super() in some cases
  constructor(props) {
    super(props);
  };

  static displayName = 'Delay';

  static defaultProps = {
    timeout: 4, // Minimum timeout is 4ms so no difference when compared with timeout zero
  };

  static propTypes = {
    timeout: PropTypes.number,
  };

  state = {
    show: false,
  };

  componentWillMount = () => {
    this.timer = null;
  };

  componentDidMount = () => {
    const { timeout } = this.props;
    this.timer = setTimeout(this.setShowValue, timeout);
    return this.timer;
  };

  componentWillUnmount = () => {
    // clear timeout_id
    this.timer ? clearTimeout(this.timer) : null;
  };

  setShowValue = () => {
    this.setState({
      show: true,
    });
  };

  render() {
    const { children } = this.props;
    const { show } = this.state;
    const performAnimation = show ? children : <div />;

    return performAnimation;
  }
}
