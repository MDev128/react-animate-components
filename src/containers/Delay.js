// @flow

import React, { PureComponent } from "react";
import PropTypes from "prop-types";

type State = {
  show: boolean
};

type DefaultProps = {
  timeout: number
};

type Props = {
  timeout: number
};

export default class Delay extends PureComponent<DefaultProps, Props, State> {
  static displayName = "Delay";

  static defaultProps = {
    timeout: 4, // Minimum timeout is 4ms so no difference when compared with timeout zero
  };

  static propTypes = {
    timeout: PropTypes.number,
  };

  state = {
    show: false,
  };

  componentWillMount = () => {
    this.timer = null;
  };

  componentDidMount = () => {
    const { timeout } = this.props;
    this.timer = setTimeout(this.setShowValue, timeout);
    return this.timer;
  };

  componentWillUnmount = () => {
    // clear timeout_id
    this.timer ? clearTimeout(this.timer) : null;
  };

  setShowValue = () => {
    this.setState({
      show: true,
    });
  };

  render(): ?React$Element<any> {
    const { children } = this.props;
    const { show } = this.state;
    const performAnimation = show ? children : <div />;

    return performAnimation;
  }
}
