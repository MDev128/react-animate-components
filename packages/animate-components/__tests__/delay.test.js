import React from 'react';
import renderer from "react-test-renderer";
import { keyframes } from "styled-components";

import Delay from '../containers/Delay';
import hoc from "../containers/HOC";

const sample = keyframes`
	0% {
		opacity: 0;
	}

	100% {
		opacity: 1;
	}
`;

const Sample = hoc("Sample", sample);

describe('Delay component', () => {
  it('should be a function', () => {
    expect(typeof Delay).toBe('function');
  });

  it('sets the timeout for an animation component', () => {
    const tree = renderer.create(
      <Delay timeout={1000}>
        <Sample duration="2s">
          Hello World!
        </Sample>
      </Delay>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
