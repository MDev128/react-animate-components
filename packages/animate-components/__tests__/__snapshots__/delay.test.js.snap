// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delay component sets the timeout for an animation component 1`] = `<div />`;
